lib/evmone/analysis.cpp

```diff
@@ -84,7 +84,8 @@ code_analysis analyze(
         auto& instr = jumpdest ? analysis.instrs.back() : analysis.instrs.emplace_back(fns[c]);
 
         auto metrics = instr_table[c];
-        block->gas_cost += metrics.gas_cost;
+        if (metrics.gas_cost > 0)  // can be -1 for undefined instruction
```

Comment:
If we know this instruction will be hit, shouldn't the cost be set to max instead (e.g. out of gas)?

Reply:
I think then it would stop with `EVMC_OUT_OF_GAS` at the beginning of the block, instead of expected `EVMC_UNDEFINED_INSTRUCTION`

Reply:
I believe we should set it to `0` as this is real gas cost of executing it. Setting it to "max" is a nice hack, but I don't waste 64-bits in the table just for it.

Reply:
https://github.com/ethereum/evmc/issues/356.